### PR TITLE
Ia color input

### PR DESCRIPTION
**Pull request by ianyimi**
color input field for media.
autosave feature from autosavebranch as well

### CODE REVIEW

File: src/tools/BuilderWand/assets/constants.ts [hunk 1]
```diff
@@ -48,3 +48,146 @@ export type HotbarGLTF = GLTF & {
     grey: THREE.MeshStandardMaterial;
   };
 };
+
+export const x11 = [
```
Comment (alex-shortt) [inline src/tools/BuilderWand/assets/constants.ts:52]: what is x11? why not COLORS or something

File: src/modifiers/BuilderMode/hooks/treeManager.ts [hunk 2]
```diff
@@ -99,17 +108,30 @@ export const useTreeManager = (): TreeManager => {
...
     refTree,
     initRefTree,
+    unsavedChanges,
+    colorHistory: colorHistory,
     hasNext: currChange.next !== undefined,
     hasPrev: currChange.prev !== undefined,
     redo,
     undo,
     add,
+    restore
```
Comment (alex-shortt) [inline src/modifiers/BuilderMode/hooks/treeManager.ts:134]: if implementation is moved to this file don't need to export this

**Review by alex-shortt — CHANGES_REQUESTED** [review]
reviewed

also, why was yarn.lock deleted?

File: src/layers/basis/index.tsx [hunk 3]
```diff
@@ -48,6 +48,7 @@ export default function Simulation(props: SimulationProps) {
 
   let CompiledWorld: ReactNode;
   if (!localWorld) {
+    //@ts-ignore
```
[the remark this answers is not in the document]
Comment (ianyimi) [reply src/layers/basis/index.tsx]: same as above

File: src/modifiers/BuilderMode/hooks/treeManager.ts [hunk 4]
```diff
@@ -99,17 +108,30 @@ export const useTreeManager = (): TreeManager => {
...
+    change.prev = currChange;
+    setCurrChange(change);
   };
 
   return {
     propsTree,
     refTree,
     initRefTree,
+    unsavedChanges,
+    colorHistory: colorHistory,
```
[the remark this answers is not in the document]
Comment (ianyimi) [reply src/modifiers/BuilderMode/hooks/treeManager.ts]: removed functionality for autosave cause i found more problems with it and the state was taking a while. will finish tuesday

**Pull request closed without merging**